### PR TITLE
Fix #443: Add support for Huawei Mobile Services

### LINKED ISSUE #443: Add support for HMS
Currently, we only support FCM by Google. We should investigate and implement HMS (Huawei Mobile Services) as well.

### PR DESCRIPTION
**Pull request by banterCZ**
- [x] Need to be tested end-to-end

### CODE REVIEW

**Review by romanstrobl — APPROVED** [review]
Looks good, I have few ideas for improvement and questions.

File: powerauth-push-server/src/main/java/io/getlime/push/service/PushMessageSenderService.java [hunk 1]
```diff
@@ -141,6 +142,15 @@ public BasePushMessageSendResult sendPushMessage(final String appId, final Mode
...
                         final PushMessageSendResult.PlatformResult platformResult = sendResult.getAndroid();
                         pushSendingWorker.sendMessageToAndroid(pushClient.getFcmClient(), pushMessage.getBody(), pushMessage.getAttributes(), pushMessage.getPriority(), token, createPushSendingCallback(mode, device, platformResult, pushMessageObject, phaser));
+                    } else if (platform == Platform.HUAWEI) {
+                        if (pushClient.getHmsClient() == null) {
+                            logger.error("Push message cannot be sent to HMS because HMS is not configured in push server.");
+                            arriveAndDeregisterPhaserForMode(phaser, mode);
+                            continue;
+                        }
+                        final String token = device.getPushToken();
+                        final PushMessageSendResult.PlatformResult platformResult = sendResult.getAndroid();
```
[the remark this answers is not in the document]
Comment (banterCZ) [reply powerauth-push-server/src/main/java/io/getlime/push/service/PushMessageSenderService.java:152]: ❗ Great catch, thanks!

File: powerauth-push-server/src/main/java/io/getlime/push/controller/rest/AdministrationController.java [hunk 2]
```diff
@@ -305,6 +311,50 @@ public Response removeAndroid(@RequestBody ObjectRequest<RemoveAndroidRequest> r
...
+     * Update Huawei configuration.
+     *
+     * @param request Update Huawei configuration request.
+     * @return Response.
+     * @throws PushServerException Thrown when application credentials entity could not be found or request validation fails.
+     */
+    @RequestMapping(value = "huawei/update", method = { RequestMethod.POST, RequestMethod.PUT })
+    public Response updateHuawei(@Valid @RequestBody ObjectRequest<UpdateHuaweiRequest> request) throws PushServerException {
+        final UpdateHuaweiRequest requestObject = request.getRequestObject();
+        logger.info("Received update Huawei request, application credentials entity ID: {}", requestObject.getAppId());
```
[the remark this answers is not in the document]
Comment (banterCZ) [reply powerauth-push-server/src/main/java/io/getlime/push/controller/rest/AdministrationController.java:324]: 🤔 A good point. I have copy&pasted the log message. This should be fixed in the android and ios methods as well.

**Comment by banterCZ** [comment]
Tested backend integration test with the mobile app with @Hopsaheysa

**Pull request merged**